### PR TITLE
Enhance pull request instructions with detailed steps

### PR DESCRIPTION
**Pull request by iAlexeze**
### SUMMARY
This pull request enhances the `CONTRIBUTING.md` file with more detailed instructions for contributors to follow before submitting a pull request. The updates aim to improve the clarity and ease of contribution to the project.

### CHANGES
  - Added detailed instructions for **code formatting** and **testing** before submitting a pull request.
  - Included links to `dune fmt` and `dune runtest` documentation for easy reference.
  - Updated the _"Set Up the Project"_ section to include clickable links for forking the repository and accessing the README.md file, enhancing navigability and ease of use.

### PURPOSE:
The purpose of these updates is to streamline the contribution process for new and/or future contributors, ensuring that their code is formatted correctly and that all tests pass before submission. This will help maintain the project's code quality and stability.

Following my first contribution experience, I was motivated to add this update to the amazing work of @yokurang on #1   with the hope that these additional information will make navigating the project more successful.

What do you think @Julow 

Thank you.

### CODE REVIEW

**Comment by yokurang** [comment]
> ### SUMMARY
> 
> This pull request enhances the `CONTRIBUTING.md` file with more detailed instructions for contributors to follow before submitting a pull request. The updates aim to improve the clarity and ease of contribution to the project.
> 
> 
> 
> ### CHANGES
> 
>   - Added detailed instructions for **code formatting** and **testing** before submitting a pull request.
> 
>   - Included links to `dune fmt` and `dune runtest` documentation for easy reference.
> 
>   - Updated the _"Set Up the Project"_ section to include clickable links for forking the repository and accessing the README.md file, enhancing navigability and ease of use.
> 
> 
> 
> ### PURPOSE:
> 
> The purpose of these updates is to streamline the contribution process for new and/or future contributors, ensuring that their code is formatted correctly and that all tests pass before submission. This will help maintain the project's code quality and stability.
> 
> 
> 
> Following my first contribution experience, I was motivated to add this update to the amazing work of @yokurang on #1   with the hope that these additional information will make navigating the project more successful.
> 
> 
> 
> What do you think @Julow 
> 
> 
> 
> Thank you.

Looks good to me 👍

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -27,7 +27,26 @@ Encountering challenges is a normal part of the process. If you need help:
...
 - **Commit Messages:** Use concise and descriptive commit messages. This helps with the project's history and review process.
-- **Pull Requests:** Before submitting a pull request, make sure your code complies with the project's standards and passes all tests by running `dune fmt` and `dune runtest`. Note you will need `ocamlformat=0.26.1`.
+- **Pull Requests:** Before submitting a pull request, make sure your code complies with the project's standards and passes all tests.. This process involves two key steps:
+  1. **Code Formatting:** Use `dune fmt` to automatically format your code according to the project's style guidelines. This command will apply the formatting rules defined in the `.ocamlformat` configuration file, ensuring consistency across the project.
+
+      To run this command, navigate to the root directory of your project in your terminal and execute:
+       
+            dune fmt
+      
+      **Note:** You will need `ocamlformat=0.26.1` installed to format your code correctly. If you haven't installed it yet, you can do so by running:
```
Comment (Julow) [inline CONTRIBUTING.md:37]: That's how it's called in Opam:

```suggestion
      **Note:** You will need `ocamlformat.0.26.1` installed to format your code correctly. If you haven't installed it yet, you can do so by running:
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -27,7 +27,26 @@ Encountering challenges is a normal part of the process. If you need help:
...
+          
+          opam install ocamlformat.0.26.1
+      
+      To learn more about `dune fmt`, check out the Man page [here](https://www.mankier.com/1/dune-fmt), or run `man dune fmt` on your terminal.
```
Comment (Julow) [inline CONTRIBUTING.md:41]: The man page is not very interesting as it contains very little documentation and a lot of unrelated options. The documentation is better: https://dune.readthedocs.io/en/stable/howto/formatting.html#running-the-formatters

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -27,7 +27,26 @@ Encountering challenges is a normal part of the process. If you need help:
...
+
+  2.  **Running Tests:** After formatting your code, run `dune runtest` to execute the project's test suite. This command will compile and run all tests to verify that your changes do not introduce any regressions.
+
+      To run this command, navigate to the root directory of your project in your terminal and execute:
+       
+            dune runtest
+      
+      To learn more about `dune runtest`, check out the Man page [here](https://www.mankier.com/1/dune-runtest), or run `man dune runtest` on your terminal.
```
Comment (Julow) [inline CONTRIBUTING.md:49]:
```suggestion
      To learn more about `dune runtest`, check out the Man page [here](https://dune.readthedocs.io/en/stable/tests.html#running-tests), or run `man dune runtest` on your terminal.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -27,7 +27,26 @@ Encountering challenges is a normal part of the process. If you need help:
 ## Contributing Guidelines
 
 - **Commit Messages:** Use concise and descriptive commit messages. This helps with the project's history and review process.
-- **Pull Requests:** Before submitting a pull request, make sure your code complies with the project's standards and passes all tests by running `dune fmt` and `dune runtest`. Note you will need `ocamlformat=0.26.1`.
+- **Pull Requests:** Before submitting a pull request, make sure your code complies with the project's standards and passes all tests.. This process involves two key steps:
+  1. **Code Formatting:** Use `dune fmt` to automatically format your code according to the project's style guidelines. This command will apply the formatting rules defined in the `.ocamlformat` configuration file, ensuring consistency across the project.
```
Comment (Julow) [inline CONTRIBUTING.md:31]: The text a bit too long and repeats `dune fmt` twice. Is it necessary to talk about `.ocamlformat` files ?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -27,7 +27,26 @@ Encountering challenges is a normal part of the process. If you need help:
...
+       
+            dune fmt
+      
+      **Note:** You will need `ocamlformat=0.26.1` installed to format your code correctly. If you haven't installed it yet, you can do so by running:
+          
+          opam install ocamlformat.0.26.1
+      
+      To learn more about `dune fmt`, check out the Man page [here](https://www.mankier.com/1/dune-fmt), or run `man dune fmt` on your terminal.
+
+  2.  **Running Tests:** After formatting your code, run `dune runtest` to execute the project's test suite. This command will compile and run all tests to verify that your changes do not introduce any regressions.
```
Comment (Julow) [inline CONTRIBUTING.md:43]: Same as above, could you make this section shorter ?

Comment (iAlexeze) [reply CONTRIBUTING.md]: Absolutely, I will get right to it.

File: CONTRIBUTING.md (hunk 4 above)
Comment (iAlexeze) [reply CONTRIBUTING.md]: Truly, there is need to talk about `.ocamlformat` here.
I will remove unnecessary information here to aid clarity.

Thank you.

File: CONTRIBUTING.md (hunk 2 above)
Comment (iAlexeze) [reply CONTRIBUTING.md]: Thanks for this.

File: CONTRIBUTING.md (hunk 3 above)
Comment (iAlexeze) [reply CONTRIBUTING.md]: This is very helpful.
Thank you.

**Comment by iAlexeze** [comment]
### Implemented suggested changes:

- Removed redundant mention of `dune fmt` and `.ocamlformat` files.
- Removed unnecessary information to aid understanding. 
- Updated links to `dune fmt` and `dune runtest` documentation for clarity.

Ready for further review.
@Julow

**Review by Julow — APPROVED** [review]
That's now perfect :) Thanks for this contribution!

I'll merge only after https://github.com/panglesd/diffcessible/pull/40 is merged as you included it in your branch. If you remove the first commit from this branch, I will be able to merge now.

**Comment by iAlexeze** [comment]
Great, thank you.
I am working on #40, and I hope to be done with it in no time.

I believe it will be better merging it together.
Thanks again for your support and guidance.

**Comment by Julow** [comment]
#40 is merged (thanks!), could you rebase this PR ?

**Comment by iAlexeze** [comment]
Done.
Please review.

@Julow 
Thank you.

**Review by Julow — APPROVED** [review]
Thanks!

**Comment by iAlexeze** [comment]
I sincerely appreciate your support and guidance. Thanks to you, I've gained valuable knowledge about OCaml and contributing to open-source projects. 

I'm eager to continue learning, growing, and making more meaningful contributions.

@Julow 
@panglesd

**Pull request merged**